src/Stack/Types/BuildPlan.hs

```diff
@@ -74,6 +74,7 @@ import           Stack.Types.Version
 -- | The name of an LTS Haskell or Stackage Nightly snapshot.
 data SnapName
     = LTS !Int !Int
+    | GHC !Int !Int !Int
```

Comment:
I'm not sure if this is the right way to go about this. Ideally, I think, the parser for `stack config set resolver` should get closer to what we have for the `resolver` field in `stack.yaml`s. So, in `Stack.ConfigCmd`, instead of using `parseSnapName` we should probably parse a `Resolver` or `AbstractResolver`, not sure about the details.